txt-suite from v2.8.0/v2.8.1 fails to execute tests

1. Verified prerequisites from: https://github.com/9elements/converged-security-suite/tree/main/cmd/core/txt-suite#prerequisites-for-usage
2. Tried release binaries from: https://github.com/9elements/converged-security-suite/releases/tag/v2.8.1 or https://github.com/9elements/converged-security-suite/releases/tag/v2.8.0

The end result on both cases looks like following:

```
root@tb:~# txt-suite exec-tests
INFO[0000] For more information about the documents and chapters, run: txt-suite -m 
INFO[0000] All tests          
INFO[0000] -------------------------------------------------- 
ReadMSR - gomsr.MSR context aborted with: too many open files
ReadMSR - gomsr.MSR context aborted with: too many open files
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/9elements/go-linux-lowlevel-hw/pkg/hwapi.IA32FeatureControlIsLocked(...)
        /home/runner/go/pkg/mod/github.com/9elements/go-linux-lowlevel-hw@v0.0.0-20240812193855-7e0a5df7e2d0/pkg/hwapi/msr_intel.go:54
github.com/9elements/converged-security-suite/v2/pkg/test.Ia32FeatureCtrl({0x8a3bd8, 0xb64240}, 0xc0000931d8?)
        /home/runner/work/converged-security-suite/converged-security-suite/pkg/test/cpu.go:230 +0xa5
github.com/9elements/converged-security-suite/v2/pkg/test.(*Test).Run(0xb307e0, {0x8a3bd8, 0xb64240}, 0xc00017f3b0)
        /home/runner/work/converged-security-suite/converged-security-suite/pkg/test/test.go:93 +0x16e
main.run({0x801d15?, 0xc000000004?}, {0xc00014e908, 0x4c, 0x792660?}, 0xc00017f3b0, 0x0)
        /home/runner/work/converged-security-suite/converged-security-suite/cmd/core/txt-suite/cmd.go:165 +0x2ab
main.(*execTestsCmd).Run(0xb43cd8, 0x2?)
        /home/runner/work/converged-security-suite/converged-security-suite/cmd/core/txt-suite/cmd.go:68 +0x5a7
reflect.Value.call({0x791dc0?, 0xb43cd8?, 0x8005a0?}, {0x801fe5, 0x4}, {0xc000010918, 0x1, 0x1?})
        /opt/hostedtoolcache/go/1.23.5/x64/src/reflect/value.go:581 +0xca6
reflect.Value.Call({0x791dc0?, 0xb43cd8?, 0xc000093bb0?}, {0xc000010918?, 0x8005a0?, 0x0?})
        /opt/hostedtoolcache/go/1.23.5/x64/src/reflect/value.go:365 +0xb9
github.com/alecthomas/kong.callMethod({0x801d42, 0x3}, {0x7cf460?, 0xb43cd8?, 0x3?}, {0x791dc0?, 0xb43cd8?, 0x0?}, 0xc00017f290)
        /home/runner/go/pkg/mod/github.com/alecthomas/kong@v0.7.1/callbacks.go:95 +0x51f
github.com/alecthomas/kong.(*Context).RunNode(0xc000134300, 0xc0000162d0, {0xc000093ed8, 0x1, 0x7bdda0?})
        /home/runner/go/pkg/mod/github.com/alecthomas/kong@v0.7.1/context.go:755 +0x825
github.com/alecthomas/kong.(*Context).Run(0x7cfe20?, {0xc000093ed8?, 0xc000093f00?, 0xc00003c613?})
        /home/runner/go/pkg/mod/github.com/alecthomas/kong@v0.7.1/context.go:780 +0x132
main.main()
        /home/runner/work/converged-security-suite/converged-security-suite/cmd/core/txt-suite/main.go:42 +0x285
```

Tried on 2 OS environments and different machines, Fedora 42 and cutom Yocto-based distro.

--- a/go.mod
+++ b/go.mod
@@ -1,9 +1,9 @@
 module github.com/9elements/converged-security-suite/v2
 
-go 1.24.0
+go 1.24.12
 
 require (
-	github.com/9elements/go-linux-lowlevel-hw v0.0.0-20240812193855-7e0a5df7e2d0
+	github.com/9elements/go-linux-lowlevel-hw v1.0.5-0.20260217195309-8b54f7428e51
 	github.com/alecthomas/kong v0.7.1
 	github.com/bxcodec/faker v2.0.1+incompatible
 	github.com/davecgh/go-spew v1.1.1
@@ -30,14 +30,14 @@ require (
 	github.com/xaionaro-go/bytesextra v0.0.0-20220103144954-846e454ddea9
 	github.com/xaionaro-go/unhash v0.0.0-20230427202706-0195a574c620
 	github.com/xaionaro-go/unsafetools v0.0.0-20210722164218-75ba48cf7b3c
-	golang.org/x/exp v0.0.0-20230626212559-97b1e661b5df
+	golang.org/x/exp v0.0.0-20240222234643-814bf88cf225
 	golang.org/x/term v0.37.0
 	gopkg.in/yaml.v3 v3.0.1
 )
 
 require (
 	github.com/DataDog/gostackparse v0.6.0 // indirect
-	github.com/dustin/go-humanize v1.0.0 // indirect
+	github.com/dustin/go-humanize v1.0.1 // indirect
 	github.com/go-ng/sort v0.0.0-20220617173827-2cc7cd04f7c7 // indirect
 	github.com/go-ng/xatomic v0.0.0-20230519181013-85c0ec87e55f // indirect
 	github.com/go-ng/xsort v0.0.0-20220617174223-1d146907bccc // indirect
@@ -48,10 +48,12 @@ require (
 	github.com/intel-go/cpuid v0.0.0-20220614022739-219e067757cb // indirect
 	github.com/jedib0t/go-pretty/v6 v6.4.6 // indirect
 	github.com/mattn/go-runewidth v0.0.13 // indirect
+	github.com/micgor32/go-msr v0.0.0-20260216140510-4af4a85b8dc7 // indirect
 	github.com/pierrec/lz4 v2.6.1+incompatible // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/rivo/uniseg v0.2.0 // indirect
 	github.com/tjfoc/gmsm v1.4.1 // indirect
+	github.com/u-root/cpuid v0.0.0 // indirect
 	golang.org/x/crypto v0.45.0 // indirect
 	golang.org/x/sync v0.18.0 // indirect
 	golang.org/x/sys v0.38.0 // indirect

--- a/pkg/hwapi/mock_pc.go
+++ b/pkg/hwapi/mock_pc.go
@@ -51,6 +51,10 @@ func (n pcmock) CPULogCount() uint32 {
 	return 0
 }
 
+func (n pcmock) CPUID(uint32, uint32) (uint32, uint32, uint32, uint32) {
+	return 0, 0, 0, 0
+}
+
 func (n pcmock) IsReservedInE820(start uint64, end uint64) (bool, error) {
 	return false, fmt.Errorf("not implemented")
 }
@@ -70,7 +74,7 @@ func (n pcmock) AddressRangesIsDMAProtected(first, end uint64) (bool, error) {
 	return false, fmt.Errorf("not implemented")
 }
 
-func (n pcmock) ReadMSR(msr int64) []uint64 {
+func (n pcmock) ReadMSR(msr int64) uint64 {
 	panic("not implemented")
 }
 
@@ -153,7 +157,7 @@ func (n pcmock) ReadHostBridgeDPR() (hwapi.DMAProtectedRange, error) {
 	return hwapi.DMAProtectedRange{}, fmt.Errorf("not implemented")
 }
 
-//MockPCReadMemory emulates a x86_64 platform memory map
+// MockPCReadMemory emulates a x86_64 platform memory map
 func MockPCReadMemory(addr uint64) byte {
 	mem := map[uint64][]byte{
 		0xFED30000: []byte{
@@ -342,7 +346,7 @@ func (n pcmock) PCIWriteConfigSpace(d hwapi.PCIDevice, off int, val interface{})
 	return fmt.Errorf("not implemented")
 }
 
-//GetPcMock returns APIInterfaces for mocking the hwapi used in unittests
+// GetPcMock returns APIInterfaces for mocking the hwapi used in unittests
 func GetPcMock(ReadMemoryFunc func(uint64) byte) hwapi.LowLevelHardwareInterfaces {
 	return pcmock{
 		ReadMemoryFunc,

--- a/pkg/provisioning/bootguard/me.go
+++ b/pkg/provisioning/bootguard/me.go
@@ -41,20 +41,17 @@ type BGInfo struct {
 // GetBGInfo reads Boot Guard msr during runtime
 func GetBGInfo(hw hwapi.LowLevelHardwareInterfaces) (*BGInfo, error) {
 	msr := hw.ReadMSR(BootGuardACMInfoMSR)
-	if msr == nil {
-		return nil, fmt.Errorf("can't read Boot Guard msr")
-	}
 	var bgi BGInfo
-	bgi.NEMEnabled = (msr[0]>>0)&1 != 0
-	bgi.TPMType = (msr[0] >> 1) & 3
-	bgi.TPMSuccess = (msr[0]>>3)&1 != 0
-	bgi.ForceAnchorBoot = (msr[0]>>4)&1 != 0
-	bgi.Measured = (msr[0]>>5)&1 != 0
-	bgi.Verified = (msr[0]>>6)&1 != 0
-	bgi.ModuleRevoked = (msr[0]>>7)&1 != 0
-	bgi.BootGuardCapability = (msr[0]>>32)&1 != 0
-	bgi.ServerTXTCapability = (msr[0]>>34)&1 != 0
-	bgi.NoResetSecretProtection = (msr[0]>>35)&1 != 0
+	bgi.NEMEnabled = (msr>>0)&1 != 0
+	bgi.TPMType = (msr >> 1) & 3
+	bgi.TPMSuccess = (msr>>3)&1 != 0
+	bgi.ForceAnchorBoot = (msr>>4)&1 != 0
+	bgi.Measured = (msr>>5)&1 != 0
+	bgi.Verified = (msr>>6)&1 != 0
+	bgi.ModuleRevoked = (msr>>7)&1 != 0
+	bgi.BootGuardCapability = (msr>>32)&1 != 0
+	bgi.ServerTXTCapability = (msr>>34)&1 != 0
+	bgi.NoResetSecretProtection = (msr>>35)&1 != 0
 	return &bgi, nil
 }
 

